### DISCUSSION TITLE
New strategy: no_prefix_except_root

### DISCUSSION THREAD

**Discussion by nazar1ua** [nuxt-modules/i18n#2058] [Ideas]
### Describe the feature

With this strategy, your routes won't have a locale prefix added except index page. The locale will be detected & changed with root language route and with i18n API.

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://v3.nuxtjs.org/community/contribution) (The contribution guideline of nuxt-modules/i18n is compliant with Nuxt too).
- [X] Check existing [discussions](https://github.com/nuxt-modules/i18n/discussions) and [issues](https://github.com/nuxt/framework/issues).

**Comment by nazar1ua** [comment]
In other words: enabling `redirectOn: 'root'` with `no_prefix` strategy

**Comment by rchl** [comment]
So are you saying that it's already possible? In which case what would be the benefit of adding new strategy?

**Comment by nazar1ua** [comment]
New strategy or allowing `redirectOn: 'root'` with `no_prefix` strategy, because for now it not working

**Comment by nazar1ua** [comment]
I want `no_prefix` for all pages except index (and couple of other pages) for SEO
Like [Trello](https://trello.com/)

**Comment by rchl** [comment]
It looks like Trello adds prefixes for all pages as long as you manually change the language in the footer.
Also it doesn't seem to automatically redirect to the user-preferred language.

**Comment by nazar1ua** [comment]
Yes, there are all pages with prefixes, but also no-prefix version with user-selected language

**Comment by nazar1ua** [comment]
@rchl Hello, I need it again, and now I can describe it from technical aspect: we need same thing as `prefix` strategy, but we need urls without prefix to work like `no_prefix` strategy, so be flexible. Basically, we need just to create strategy (as I understood, in [intlify/routing](https://github.com/intlify/routing) repository) but not to override nuxt default routes, instead just add routes with prefixes on top of nuxt default routes.

**Comment by nazar1ua** [comment]
@kazupon @BobbieGoede
